### PR TITLE
Change the view for instructors to show all students

### LINKED ISSUE #103: Bug displaying number of challenges marked as complete
Looks as though (possibly) the total number of exercises marked by everyone on the course is being displayed, rather than just my own?

image

Not critical to get solved immediately, but I don't think this was a problem yesterday.

### PR DESCRIPTION
**Pull request by alasdairwilson**
closes #103 #78 

Both indicate the same issue, that for instructors the problem solved count doesn't display correctly. This is because instructors get all problems done by users of type "student" when they make a eventid/problem api call whereas students only get their own problems.

Instead of fixing this, I have changed the logic such that for instructors, the total displayed is [total solved problems by all students]/[number of problems in section * uniqueusers]

image
image

### CODE REVIEW

File: components/EventItemView.tsx [hunk 1]
```diff
@@ -44,10 +49,25 @@ const EventView: React.FC<EventItemProps> = ({ material, item, problems }) => {
...
-    completedLabel = `[${completedProblems.length}/${itemProblems.length}]`;
-    isCompleted = completedProblems.length === itemProblems.length;
+    if (uniqueUsersCount > 1) {
+      // if you are an instructor then you get a total count of problems completed / total problems available to all students
+      completedLabel = `[${completedProblems.length}/${itemProblems.length * uniqueUsersCount}]`
+      isCompleted = completedProblems.length === (itemProblems.length * uniqueUsersCount);
+    }
+    else {
+      // if you are a student then you get a count of problems completed / total problems available to just you
+      completedLabel = `[${completedProblems.length}/${itemProblems.length}]`
```
Comment (martinjrobins) [inline components/EventItemView.tsx:68]: minor point, but isn't this branch just the same as the first with `uniqueUsersCount = 1`? Could combine to:

```typescript
uniqueUsersCount = max(1, uniqueUsers.size); // to deal with =0
completedLabel = `[${completedProblems.length}/${itemProblems.length * uniqueUsersCount}]`
isCompleted = completedProblems.length === (itemProblems.length * uniqueUsersCount);
```

**Review by martinjrobins — APPROVED** [review]
looks great @alasdairwilson, minor point below

File: components/EventItemView.tsx (hunk 1 above)
Comment (alasdairwilson) [reply components/EventItemView.tsx]: well spotted!

**Review by martinjrobins — APPROVED** [review]
looks all good, will merge :)

**Pull request merged**